### DISCUSSION TITLE
Deploy Rustdesk Over Windows Domain Clients

### DISCUSSION THREAD

**Discussion by billgatos** [rustdesk/rustdesk#1865] [General]
Hi all, 

In our environment we used Anydesk, and so now its going to be Rustdesk, to be used as a helper to our users.
We managed so far to create a setup install script to install silently the Rustdesk on every PC, but the settings regardind the internal host and key is still a issue.

Our script in the startup GPo is like this:
**

REM setup RustDesk

@echo off

wmic product get name,version /format:csv | find "RustDesk"
if %errorlevel%==0 (
goto:1
) else (
goto:2
)

:1
echo Rustdesk instalado

goto:eof

:2
echo Setting up Remoto RustDesk

\\localserver.domain.local\c$\IT\Software\RustDesk\rustdesk-1.1.9-putes.exe --silent-install 1

goto:eof

:eof

** 
We have tried to copy the TOML file but is replaced by an empty one, if we send the exe renamed it need to install and UAC doesn't work.

Is there a way to install, with already defined host and key values.

best regards.

**Comment by rustdesk** [comment]
https://github.com/rustdesk/rustdesk/issues/769

**Comment by billgatos** [comment]
i also tried after install the following:
at the installation folder on a new client i tried 
rustdesk.exe --import-config "......rustdesk2.toml" where toml file has inside the key, and the server ip, but that config is not imported also to the new client.

**Comment by cableTh0rn** [comment]
Hello , we have successfully tested deployment via gpo. Create a new GPO and place the Install powershell script as a computer startup script. We added some extra commands to the script to set the rustdesk_ID to the hostname. The key and other config is in the powershell script . No need to write extra script to deploy the software to clients silently.

**Reply by emedeirosit** [reply]
Can you please tell me what parameters did you use to set host and key? I mean: if you used command line options: maybe you modified the name of the executable?

**Reply by ghost** [reply]
Can you share the script please? (minus sensitive details obviously).

**Reply by cableTh0rn** [reply]
After server installation (self-host) Rustdesk creates an client installation powershell script for windows and a shell script for  linux clients. This script can be downloaded from http://server_hostname_or_ip:8000/ The scripts already contain key, host and other config parameters. But can also be changed afterwards. When run, script downloads the latest rustdesk client (executable file) and installs it. You can deploy clients via GPO computer startup script.

**Reply by billgatos** [reply]
Regarding this procedure, i can't reach on the server wich is a windows server the site http://....:8000 
Maybe going to try the build sugestion

**Comment by emedeirosit** [comment]
Have you considered making your own build? I don't know the first thing about using Git or any of the development tools involved, but yesterday I successfully made my first build from the nightly release, hardcoding the values for host and key, and it was really easy.

https://rustdesk.com/docs/en/dev/build/all/

The only issue for me is that the browsers try to block the download of this custom installer, and it can be a problem when someone is letting me access their computer for the first time. In your case it won't matter, since you control the entire environment.
